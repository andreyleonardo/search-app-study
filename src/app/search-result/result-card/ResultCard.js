import React from 'react';
import PropTypes from 'prop-types';
import ResultCardTitle from './result-card-title/ResultCardTitle';
import ResultCardDetails from './result-card-details/ResultCardDetails';

import './ResultCard.scss';

const ResultCard = ({
  index, title, rating, currency, price, isSpecialOffer, onClick, tabIndex, imgSrc, imgAlt
}) => (
  <div
    id={`result-card-id-${index}`}
    className="result_card row"
    role="button"
    onKeyPress={onClick}
    onClick={onClick}
    tabIndex={tabIndex}
  >
    <picture className="result_card__picture col-xs-12 col-sm-3">
      <img className="result_card__img" src={imgSrc} alt={imgAlt} />
    </picture>
    <div className="result_card__details col-xs-12 col-sm">
      <ResultCardTitle title={title} />
      <ResultCardDetails
        rating={rating}
        currency={currency}
        price={price}
        isSpecialOffer={isSpecialOffer}
      />
    </div>
  </div>
);

ResultCard.propTypes = {
  title: PropTypes.string.isRequired,
  rating: PropTypes.number,
  currency: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  isSpecialOffer: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  tabIndex: PropTypes.number.isRequired,
  index: PropTypes.number,
  imgAlt: PropTypes.string,
  imgSrc: PropTypes.string
};

ResultCard.defaultProps = {
  rating: 0,
  isSpecialOffer: false,
  imgAlt: '',
  imgSrc: '',
  index: 0
};

export default ResultCard;
