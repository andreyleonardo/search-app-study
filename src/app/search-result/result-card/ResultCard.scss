.result_card {
  cursor: pointer;
  margin: 3px;
  padding: 0.2em;
  background-color: white;
  min-height: 100px;
  transition: box-shadow 0.5s;
  text-align: left;

  &:hover {
    box-shadow: 0px 0px 2px 5px rgba(255,255,255,1);
  }

  &:focus {
    outline: none;
  }

  &__details {
    padding: 10px;
  }

  &__img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  &__picture {
    padding: 10px;
  }
}
