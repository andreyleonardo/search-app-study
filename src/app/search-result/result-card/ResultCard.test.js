import React from 'react';
import ResultCard from './ResultCard';

describe('Result Card', () => {
  let resultCard;
  let props;

  beforeEach(() => {
    props = {
      title: 'Title',
      currency: '$',
      price: '30',
      rating: 4.8,
      onClick: jest.fn(),
      tabIndex: 1
    };

    resultCard = shallow(<ResultCard {...props} />);
  });

  it('renders without crashing', () => {
    expect(resultCard).toMatchSnapshot();
  });

  it('calls onClick when component is clicked', () => {
    resultCard.simulate('click');

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
