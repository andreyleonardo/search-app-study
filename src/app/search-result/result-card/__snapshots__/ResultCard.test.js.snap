// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Card renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultCard
    currency="$"
    imgAlt=""
    imgSrc=""
    index={0}
    isSpecialOffer={false}
    onClick={[MockFunction]}
    price="30"
    rating={4.8}
    tabIndex={1}
    title="Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <picture
          className="result_card__picture col-xs-12 col-sm-3"
        >
          <img
            alt=""
            className="result_card__img"
            src=""
          />
        </picture>,
        <div
          className="result_card__details col-xs-12 col-sm"
        >
          <ResultCardTitle
            title="Title"
          />
          <ResultCardDetails
            currency="$"
            isSpecialOffer={false}
            price="30"
            rating={4.8}
          />
        </div>,
      ],
      "className": "result_card row",
      "id": "result-card-id-0",
      "onClick": [MockFunction],
      "onKeyPress": [MockFunction],
      "role": "button",
      "tabIndex": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            className="result_card__img"
            src=""
          />,
          "className": "result_card__picture col-xs-12 col-sm-3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "result_card__img",
            "src": "",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "picture",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ResultCardTitle
              title="Title"
            />,
            <ResultCardDetails
              currency="$"
              isSpecialOffer={false}
              price="30"
              rating={4.8}
            />,
          ],
          "className": "result_card__details col-xs-12 col-sm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currency": "$",
              "isSpecialOffer": false,
              "price": "30",
              "rating": 4.8,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <picture
            className="result_card__picture col-xs-12 col-sm-3"
          >
            <img
              alt=""
              className="result_card__img"
              src=""
            />
          </picture>,
          <div
            className="result_card__details col-xs-12 col-sm"
          >
            <ResultCardTitle
              title="Title"
            />
            <ResultCardDetails
              currency="$"
              isSpecialOffer={false}
              price="30"
              rating={4.8}
            />
          </div>,
        ],
        "className": "result_card row",
        "id": "result-card-id-0",
        "onClick": [MockFunction],
        "onKeyPress": [MockFunction],
        "role": "button",
        "tabIndex": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              className="result_card__img"
              src=""
            />,
            "className": "result_card__picture col-xs-12 col-sm-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "result_card__img",
              "src": "",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "picture",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ResultCardTitle
                title="Title"
              />,
              <ResultCardDetails
                currency="$"
                isSpecialOffer={false}
                price="30"
                rating={4.8}
              />,
            ],
            "className": "result_card__details col-xs-12 col-sm",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "Title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "$",
                "isSpecialOffer": false,
                "price": "30",
                "rating": 4.8,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
